docs/design/security_context.md

```diff
@@ -103,88 +103,55 @@ If the value of the SecurityContextProvider field on the Kubelet is nil, the kub
-	AllowedVolumeTypes []string
-	
-	// AddCapabilities is the list of Linux kernel capabilities to add
-	AddCapabilities []string
-	
-	// RemoveCapabilities is the list of Linux kernel capabilities to remove
-	RemoveCapabilities []string
-	
-	// Isolation specifies the type of isolation required for containers 
-	// in this security context 
-	Isolation ContainerIsolationSpec
-}
+	// Capabilities are the capabilities to add/drop when running the container
+	Capabilities *Capabilities
 
-// ContainerIsolationSpec indicates intent for container isolation
-type ContainerIsolationSpec struct {
-	// Type is the container isolation type (None, Private)
-	Type ContainerIsolationType
-	
-	// FUTURE: IDMapping specifies how users and groups from the host will be mapped
-	IDMapping *IDMapping
-}
+	// Run the container in privileged mode
+	Privileged *bool
 
-// ContainerIsolationType is the type of container isolation for a security context
-type ContainerIsolationType string
+	// SELinuxOptions are the labels to be applied to the container
+	// and volumes
+	SELinuxOptions *SELinuxOptions
 
-const (
-    // ContainerIsolationNone means that no additional consraints are added to
-    // containers to isolate them from their host
-	ContainerIsolationNone ContainerIsolationType = "None"
-	
-	// ContainerIsolationPrivate means that containers are isolated in process
-	// and storage from their host and other containers.
-	ContainerIsolationPrivate ContainerIsolationType = "Private"
-)
-
-// IDMapping specifies the requested user and group mappings for containers 
-// associated with a specific security context
-type IDMapping struct {
-	// SharedUsers is the set of user ranges that must be unique to the entire cluster
-	SharedUsers []IDMappingRange
-	
-	// SharedGroups is the set of group ranges that must be unique to the entire cluster
-	SharedGroups []IDMappingRange
+	// RunAsUser is the UID to run the entrypoint of the container process.
+	RunAsUser *int64
+}
 
-	// PrivateUsers are mapped to users on the host node, but are not necessarily
-	// unique to the entire cluster
-	PrivateUsers []IDMappingRange
+// SELinuxOptions are the labels to be applied to the container.
+type SELinuxOptions struct {
+	// SELinux user label
+	User string
 
-	// PrivateGroups are mapped to groups on the host node, but are not necessarily
-	// unique to the entire cluster
-	PrivateGroups []IDMappingRange
-}
+	// SELinux role label
+	Role string
 
-// IDMappingRange specifies a mapping between container IDs and node IDs
-type IDMappingRange struct {
-	// ContainerID is the starting container UID or GID
-	ContainerID int
+	// SELinux type label
+	Type string
 
-	// HostID is the starting host UID or GID
-	HostID int
-	
-	// Length is the length of the UID/GID range
-	Length int
+	// SELinux level label.
+	Level string
 }
-
 ```
+### Admission
 
+It is up to an admission plugin to determine if the security context is acceptable or not.  At the
```

Comment:
We decided to go with a single admission plugin for now https://github.com/GoogleCloudPlatform/kubernetes/pull/7343#discussion_r29300725